### ISSUE TITLE
wesnoth.units.to_map crashes game when used twice

### ISSUE THREAD

**Issue by ProditorMagnus** [wesnoth/wesnoth#8229]
### Game and System Information

- **Version:** 1.17.24

### Description of the bug

Game crashes and log ends with 20240109 01:04:27 error scripting/lua: Could not find unit 2 on the map

### Steps to reproduce the behavior

1. ~add-ons/EventLoader/action.cfg =

```
# lua wesnoth.require("wml-utils").handle_event_commands(wml.load("~add-ons/EventLoader/action.cfg"))
[lua]
code=<<
u = wesnoth.units.find_on_map({side=2})[1]
u:extract()
u:to_map(false)
u:to_map(false)
>>
[/lua]
```
2. :alias a=lua wesnoth.require("wml-utils").handle_event_commands(wml.load("~add-ons/EventLoader/action.cfg"))
3. :a

### Expected behavior

No crash.

### Additional context

Encountered when debugging https://github.com/wesnoth/wesnoth/blob/6a280a8a1ae8e94708f3d095dfa8018536492387/data/test/scenarios/wml_tests/ScenarioWML/EventWML/ActionWML/DirectActionsWML/harm_unit_survivable_experience_no.cfg based on https://github.com/wesnoth/wesnoth/blob/6a280a8a1ae8e94708f3d095dfa8018536492387/data/lua/wml/harm_unit.lua

**Comment by gfgtdf** [comment]
The problem is that this line https://github.com/wesnoth/wesnoth/blob/3804027d100fc02328a6829997792b8ef0587303/src/scripting/game_lua_kernel.cpp#L2453
removes the unit, i actually don't know why it does that when lua_unit::put_map also does that.

**Comment by stevecotton** [comment]
Fixed in #8239.
